src/python/pants/engine/internals/selectors.py

```diff
@@ -72,58 +70,57 @@ def parse_product_and_subject_types(
                 f"Expected either two or three arguments, but got {len(get_args)} arguments."
             )
 
-        product_expr = get_args[0]
-        if not isinstance(product_expr, ast.Name):
+        output_expr = get_args[0]
+        if not isinstance(output_expr, ast.Name):
             raise parse_error(
-                "The first argument should be the type of the product, like `Digest` or "
-                "`ProcessResult`."
+                "The first argument should be the output type, like `Digest` or `ProcessResult`."
             )
-        product_type = product_expr.id
+        output_type = output_expr.id
 
-        subject_args = get_args[1:]
-        if len(subject_args) == 1:
-            subject_constructor = subject_args[0]
-            if not isinstance(subject_constructor, ast.Call):
+        input_args = get_args[1:]
+        if len(input_args) == 1:
+            input_constructor = input_args[0]
+            if not isinstance(input_constructor, ast.Call):
                 raise parse_error(
-                    "Because you are using the shorthand form Get(ProductType, "
-                    "SubjectType(constructor args), the second argument should be a constructor "
+                    "Because you are using the shorthand form Get(OutputType, "
+                    "InputType(constructor args), the second argument should be a constructor "
```

Comment:
Perhaps consider generating at least this much with an inner method or something? Also very optional, but I find myself doing this a lot with parsers and I believe reducing the chances of misspelling something in these difficult error messages is useful.
```python
def parse_get_shorthand_error(second_arg_should_be: String) -> String:
  return dedent(f"""\
    Because you are using the shorthand form Get(OutputType, \
    InputType(constructor args), the second argument should be a {second_arg_should_be}\
    """
```

Reply:
I'd agree if there were >2 of these messages. Because it's only 2, and they are right next to each other, I think it's less cognitive load to have a tiny bit of duplication.